File: solutions/fsharp/armstrong-numbers/1/ArmstrongNumbers.fs
module ArmstrongNumbers

open System

let asDigitArray n =
    n.ToString().ToCharArray()
    |> Array.map (Char.GetNumericValue >> int)

let isArmstrongNumber (number: int): bool = 
    let numbers = number |> asDigitArray
    number = ((0, numbers) ||> Array.fold (fun acc curr -> acc + pown curr numbers.Length))

File: solutions/csharp/armstrong-numbers/1/ArmstrongNumbers.cs
using System;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        var numbers = number.ToString().Select(char.GetNumericValue).ToArray();

        return numbers.Sum(n => Math.Pow(n, numbers.Length)) == number;
    }
}